### ISSUE TITLE
Models without fields are serialized to None

### ISSUE THREAD

**Issue by magaman384** [schematics/schematics#235]
Hello,
I'm not sure if it is bug or feature but following model will be converted to None after calling serialize:
```
class EmptyModel(Model):
    pass

model = EmptyModel({})
model.serialize() # this returns None instead of {}
```

**Comment by bintoro** [comment]
Contrary to the previous comment, this is not the same as the "unset field" issue, although it's related. This here is about the serialized representation of empty model instances.

An empty instance will currently be exported as `None`. OP expected to see `{}`.

An instance is empty if
* its class doesn't define any fields (OPs example) or
* none of its fields contains a nonempty value AND `serialize_when_none=False`.

```python
class Test(Model):

    class Options:
        serialize_when_none = False

    l = ListType(IntType())
    s = StringType()
```
```python
>>> t = Test()
>>> t.l = []
>>> print(t.to_primitive())
None
```

The behavior is consistent with `ModelType` fields:


```python
class Outer(Model):
    m = ModelType(Test)
```
```python
>>> o = Outer()
>>> o.m = Test()
>>> print(o.to_primitive())
{'m': None}
```

The question is: is `None` the ideal representation for empty models?

Well, I think `None` makes perfect sense for now, and I don't know how `{}` would be an improvement.

The only argument for `{}` that I can think of is that the current behavior is not entirely consistent between `ModelType` and, e.g., `ListType`:
* `ListType` accepts both `None` and `[]`, and they're serialized as such.
* `ModelType` accepts both `None` and `Foo()`, but they're both serialized as `None`.

My take on this is that the `ModelType` behavior is actually desirable, and it's `ListType` that's acting strange. The fact that `ListType` can already accommodate two different kinds of "empty" means that, once the Undefined feature lands, there will be three kinds.

Therefore, the present issue seems to need addressing only if

* there's currently a scenario where a model or a `ModelType` field will get serialized as `{}`, as that would be inconsistent, or

* there would be some kind of semantic difference between `None` and `{}`, but I don't readily see what that might be.

**Comment by bintoro** [comment]
Let me revise my previous statement.

Obviously, there is a difference between `obj.field = None` and `obj.field = ModelType(EmptyModel)`. The latter is an actual object even if it has nothing to export.

It's just that, under default settings, this is such an artificial scenario so as not to matter.

However, the `serialize_when_none` switch will de facto turn any instance into an `EmptyModel` if it doesn't contain at least one value. This is a common occurrence and actually pretty bizarre if you think about it.

I mean, it's fine to have an option for filtering out empty values, but that process shouldn't be able to reduce something to nothing.

(As a database analogy, a foreign key reference doesn't magically become null just because the referenced row is updated to only contains nulls.)

So I guess `{}` would in fact be the most logical representation for a `ModelType` field that contains an instance but has no values to export. This would distinguish it from a field that truly contains `None`.

**Comment by bintoro** [comment]
Fixed by PR #351.
